API cleanup

dthaler opened the pull request:
1. Move timed_execution() into the subdirectory for the check command.

2. Clean up global_options_t by making options be passed in as a parameter, rather than relying on a global variable, and rename the type accordingly. Allow passing NULL (for C compatibility) as a shortcut to just use the default options.

3. Rather than always writing to std::cout, allow the output stream to be passed in as an argument. (This PR currently doesn't affect CRAB_LOG.)

4. Add ebpf_verifier.hpp to simplify writing apps, so one only can just include the master header to use the static lib, instead of having to keep track of what header goes with what API.

elazarg commented on src/crab_verifier.cpp:
```diff
@@ -99,43 +101,33 @@ static checks_db generate_report(cfg_t& cfg,
-        print_report(report);
+    if (options->print_failures) {
+        print_report(s, report);
     }
```
Printing is the main reason I put the measurements inside this function: it's meaningless to have both verbose mode and measurements. I'm not sure that's important though, since measurements are realtively consistent across runs.

dthaler replied:
That's what I concluded too.  Is there a change you'd like me to make or is it ok?

elazarg replied:
No, I think that's ok.

elazarg commented:
Is this ready for merge? The commit message still says WIP (of course I can change that during squash).

dthaler commented:
> Is this ready for merge? The commit message still says WIP (of course I can change that during squash).

Ha, yes, I just squashed and updated the commit message :)
